Fixed #4872 - Fix some async subpanel actions failing if refresh_page=1

lazka opened the pull request:
## Description

In case the subpanel layout uses refresh_page=1 then showSubPanel() will start an
async request and after that reload the page. In many cases this will lead to the
just started request being aborted and the page gets reloaded without anything changing.

This was introduced in 829a06f5b8af where the previously sync call was replaced with an
async one.

This is most visible for me in the "Users" module were I can't remove security groups
through the subpanel and it just refreshes the page without removing the group.

Fix this my moving the page refresh check in the ajax callback so the page gets only reloaded
once the request is finished.

## Motivation and Context

Make adding and removing security groups and roles in the Users module work
for example.

## How To Test This

(Using Firefox, in case that makes a difference)

1) Admin -> User Management
2) Select a user
3) Scroll down to the security groups subpanel
4) Add a security group
5) Click "Edit" on the newly added group and then "Remove"

Without this patch the page reloads and the security group is still there.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

lazka commented:
@PedroErnst since you made the async change, could you have a quick look maybe? thanks!

pgorod commented:
Is this the same Issue as #4872 ?

In one of the comments a user claims he can't delete security groups from the user's detail view.

pgorod commented:
@lazka it seems your patch solves the problem mentioned in a comment in #4872.

If you can add a fix for the PHP Warnings that would be great, we could close that Issue. It's normally very easy, it's just a matter of matching parameter definitions. I am assuming you get the same errors on your installation?

lazka commented:
> @lazka it seems your patch solves the problem mentioned in a comment in #4872.

Thanks. I've added `Fixed #4872` to the first commit and the PR title.

> If you can add a fix for the PHP Warnings that would be great, we could close that Issue. It's normally very easy, it's just a matter of matching parameter definitions. I am assuming you get the same errors on your installation?

I've added a second commit fixing the warning.

Dillon-Brown commented:
Assessed
